WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -240,18 +253,22 @@ class PostCoordinator: NSObject {
         backgroundService.refreshPostStatus()
     }
 
-    private func upload(post: AbstractPost) {
+    private func upload(post: AbstractPost, completion: ((Result<AbstractPost>) -> ())? = nil) {
         mainService.uploadPost(post, success: { uploadedPost in
             print("Post Coordinator -> upload succesfull: \(String(describing: uploadedPost.content))")
 
             SearchManager.shared.indexItem(uploadedPost)
 
             let model = PostNoticeViewModel(post: uploadedPost)
             ActionDispatcher.dispatch(NoticeAction.post(model.notice))
+
+            completion?(.success(uploadedPost))
         }, failure: { error in
             let model = PostNoticeViewModel(post: post)
             ActionDispatcher.dispatch(NoticeAction.post(model.notice))
 
+            completion?(.error(error ?? SavingError.unknown))
```

Comment:
What are we actually doing with this error? Can we send more info here?

Reply:
Hmm, I'm not sure I got your question.

Here we're sending the error through a block. If the `error` is `nil` then we send an `unknown` error, because we don't know what happened. Otherwise, the `error` contains all relevant data about what happened.

If that doesn't answer your question please let me know.